litho-kotlin/src/main/java/com/facebook/litho/kotlin/LithoKotlinDsl.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2014-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.litho.kotlin
+
+import com.facebook.litho.*
+import com.facebook.litho.sections.widget.RecyclerCollectionComponent
+import com.facebook.litho.widget.*
+import java.util.*
+
+/** A concrete class to help scope the DSL. */
+class ChildHolder(private val context: ComponentContext) {
+  private val children = ArrayList<Component.Builder<*>>()
+
+  fun <B : Component.ContainerBuilder<B>> apply(containerBuilder: Component.ContainerBuilder<B>) {
+    children.forEach { containerBuilder.child(it) }
+  }
+
+  /**
+   * Base implementation for components as children of containers.
+   * @param create the creator function reference. E.g. Text::create.
+   * @param init the DSL builder lambda.
+   */
+  fun <B : Component.Builder<B>> componentBuilder(
+      create: (c: ComponentContext) -> B,
+      init: B.() -> Unit) {
+    val builder = create(context)
+    builder.init()
+    children.add(builder)
+  }
+
+  fun column(init: Column.Builder.() -> Unit) = componentBuilder(Column::create, init)
+  fun row(init: Row.Builder.() -> Unit) = componentBuilder(Row::create, init)
```

Comment:
I wish we had a way, where this pattern could be abstracted i.e. 
`component(init: Component.Builder.() -> Unit) = componentBuilder(Component::create, init)`

I can't think of one since they are not sharing a common interface, but if you find one such solutions it would make me super happy :D